fxos-tv-modal-dialog.js

```diff
@@ -132,20 +143,23 @@
   };
 
   proto._open = function(options) {
-    this._focusedIndex = this.defaultFocusIndex;
     this._renderMessage(options);
     this._renderButtons(options);
     BaseModalDialog.prototype._open.call(this, options);
   };
 
   proto.startKeyNavigation = function() {
-    this.element.addEventListener('keydown', this);
-    if (this.buttonElements.length > 0) {
-      if (this._focusedIndex === -1) {
-        this._focusedIndex = 0;
-      }
-      this.focus();
-    }
+    var id = '#' + this.element.id;
+    SpatialNavigation.init();
+    SpatialNavigation.add({
+      id: this.element.id,
+      selector: id + ' button, ' + id + ' input, ' + id + ' fxos-tv-button',
+      // enterTo: 'last-focused',
+      restrict: 'self-only'
+    });
+
+    SpatialNavigation.makeFocusable();
+    SpatialNavigation.focus(this.defaultFocusElement);
```

Comment:
The first navigable element will be focused if `this.defaultFocusElement` is `null`. I just want to confirm if that's what you expect.